update.cf

```diff
@@ -43,19 +43,61 @@ body agent control
 
 bundle common update_def
 {
-  vars:
+  classes:
+      # all extra classes are defined true
+      "$(extra_classes)" expression => "any", meta => { "extra" };
+
+      # all override classes are defined true
+      "$(override_classes)" expression => "any", meta => { "override" };
+
+      "have_augments_file" expression => fileexists($(augments_file)), scope => "bundle";
 
+      "augments_are_json" expression => regcmp(".+\.json", $(augments_file)), scope => "bundle";
+      "have_json_augments" and => { "have_augments_file", "augments_are_json" }, scope => "bundle";
+
+      "augments_are_yaml" expression => regcmp(".+\.yaml", $(augments_file)), scope => "bundle";
+      "have_yaml_augments" and => { "have_augments_file", "augments_are_yaml" }, scope => "bundle";
+
+  vars:
       "current_version" string => "3.7.0";
+    any::
+      "augments_file" string => ifelse(fileexists("$(this.promise_dirname)/services/def.yaml"), "$(this.promise_dirname)/services/def.yaml",
+                                      "$(this.promise_dirname)/services/def.json");
+
+      "defvars" slist => variablesmatching("default:def\..*", "defvar");
```

Comment:
Yep, it needs to be "default:update_def\..*" instead of default:def\..*. I dont get the masterfiles_perms_mode printed out unless its update_def.
 
R: update_def: default:update_def.masterfiles_perms_mode = 0600